pyartcd/pyartcd/tekton.py

```
@@ -0,0 +1,148 @@
+    """Annotate the current PipelineRun with the given key-value pairs.
+
+    Silently skips if not running inside a Tekton context or if the
+    PipelineRun name is unavailable.
+    """
+    pr_name = get_current_pipelinerun_name()
+    if not pr_name:
+        return
+    ns = _get_namespace()
+    for key, value in annotations.items():
+        try:
+            subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)